src/test/java/org/opentripplanner/updater/vehicle_positions/RealtimeVehiclePositionTest.java

```diff
@@ -0,0 +1,100 @@
+            List<RealtimeVehiclePosition> vehiclePositions = pattern.getVehiclePositions();
+            realtimeVehicleCount.addAndGet(vehiclePositions.size());
+        });
+
+        return realtimeVehicleCount.get();
+    }
+
+    @Test
+    public void testRealtimeTest() throws Exception {
+        // Initialize updater
+        GtfsRealtimeFileVehiclePositionSource vehiclePositionSource = new GtfsRealtimeFileVehiclePositionSource();
+
+        // Create config
+        JsonNode config = this.generateConfig("kcm_rt_gtfs.pb");
+
+        // Process realtime feed
+        vehiclePositionSource.configure(this.graph, config);
+        List<GtfsRealtime.VehiclePosition> positions = vehiclePositionSource.getPositions();
+
+        // Ensure feed was parsed
+        Assert.assertNotNull(positions);
+
+        // Map positions to trips in feed
+        VehiclePositionPatternMatcher vehiclePositionPatternMatcher = new VehiclePositionPatternMatcher(this.graph);
+        vehiclePositionPatternMatcher.applyVehiclePositionUpdates(positions, this.feedId.getId());
+
+        // Get all patterns in feed
+        Collection<TripPattern> patterns = this.graph.index.patternsForFeedId.get(this.feedId.getId());
+
+        // Count number of vehicles across all patterns
+        int realtimeVehicleCount = this.getRealtimeVehicleCountForPatterns(patterns);
+
+        Assert.assertEquals(627, realtimeVehicleCount);
+
+        // Update positions
+        JsonNode updated_config = this.generateConfig("kcm_rt_gtfs_2.pb");
+
+        // Process realtime feed
+        vehiclePositionSource.configure(this.graph, updated_config);
+        List<GtfsRealtime.VehiclePosition> updated_positions = vehiclePositionSource.getPositions();
+
+        // Ensure feed was parsed and that it was correctly different
+        assertNotEquals(positions.size(), updated_positions.size());
+
+        // In the test data, updated_positions is smaller
+        for (int i = 0; i < updated_positions.size(); i++) {
+            Assert.assertNotEquals(positions.get(i), updated_positions.get(i));
+        }
+
+        // Execute the same update step as the runner
+        vehiclePositionPatternMatcher.wipeSeenTripIds();
+        vehiclePositionPatternMatcher.applyVehiclePositionUpdates(updated_positions, this.feedId.getId());
+
+        // Ensure that without cleaning the number of active vehicles is wrong
+        int updatedRealtimeVehicleCount = getRealtimeVehicleCountForPatterns(patterns);
+        Assert.assertEquals(1016, updatedRealtimeVehicleCount);
+
+        // "clean" patterns
+        vehiclePositionPatternMatcher.cleanPatternVehiclePositions(this.feedId.getId());
+
+        // Ensure that after cleaning it is correct
+        updatedRealtimeVehicleCount = getRealtimeVehicleCountForPatterns(patterns);
+        Assert.assertEquals(updated_positions.size(), updatedRealtimeVehicleCount);
+    }
+
+    @Override
+    public final String getFeedName() {
+        return "kcm_rt_gtfs.zip";
```

Comment:
I gave this a shot, but there was no existing feed that had a matching realtime feed. I also tried removing unneeded parts of the kcm zip file, but this failed validation and OTP refused to import it. I think the only solution to reduce the file size would be to manually strip out routes, trips and stops not seen in the realtime data. I am not sure how to best go about doing this/if it is worth the time

Reply:
Yes, I think it would be a good idea to just strip all routes/trips/stops/stop_times not used by tests (and potentially do the same to the rt feed).

Reply:
I had to write a custom python script to do it, but the test zip file has been successfully shrunk in 0de99f5c9fb0db212b461c57c3861eccb43bc3b3! Do you think it's worth trying to wipe the history with https://rtyley.github.io/bfg-repo-cleaner/ to make use of the savings/make a new commit?

Reply:
Whoa that's really cool. Didn't know about that tool. Yea, might be worth it. cc @binh-dam-ibigroup, this might be useful for cleaning up https://github.com/ibi-group/datatools-ui/pull/598

Reply:
This might be a big enough/dangerous enough action that it makes more sense to do on a call. Also I think the tool only works on the HEAD branch so will have to wait until after merge.